warehouse/bigquery/bigquery_source_test.go

```diff
@@ -4,162 +4,82 @@ package bigquery_test
 
 import (
 	"context"
-	"encoding/json"
 	"fmt"
-	"log"
 	"os"
 	"testing"
 
 	"github.com/rudderlabs/rudder-server/utils/misc"
 
 	"github.com/stretchr/testify/require"
 
-	"cloud.google.com/go/bigquery"
-
 	bigquery2 "github.com/rudderlabs/rudder-server/warehouse/bigquery"
 	"github.com/rudderlabs/rudder-server/warehouse/client"
 	"github.com/rudderlabs/rudder-server/warehouse/testhelper"
 	warehouseutils "github.com/rudderlabs/rudder-server/warehouse/utils"
-	"github.com/stretchr/testify/assert"
 )
 
-type TestHandle struct {
-	DB             *bigquery.Client
-	WriteKey       string
-	Schema         string
-	Tables         []string
-	SourceId       string
-	DestinationId  string
-	SourceWriteKey string
-}
-
-var handle *TestHandle
+func TestSourcesBigQueryIntegration(t *testing.T) {
+	t.Parallel()
 
-const (
-	TestCredentialsKey = testhelper.BigqueryIntegrationTestCredentials
-	TestSchemaKey      = testhelper.BigqueryIntegrationTestSchema
-)
-
-// bigqueryCredentials extracting big query credentials
-func bigqueryCredentials() (bqCredentials bigquery2.BQCredentialsT, err error) {
-	cred, exists := os.LookupEnv(TestCredentialsKey)
-
-	if !exists {
-		err = fmt.Errorf("following %s does not exists while running the Bigquery test", TestCredentialsKey)
-		return
+	if _, exists := os.LookupEnv(testhelper.BigqueryIntegrationTestCredentials); !exists {
+		t.Skipf("Skipping %s as %s is not set", t.Name(), testhelper.BigqueryIntegrationTestCredentials)
 	}
 
-	err = json.Unmarshal([]byte(cred), &bqCredentials)
-	if err != nil {
-		err = fmt.Errorf("error occurred while unmarshalling bigquery test credentials with err: %s", err.Error())
-		return
-	}
-	return
-}
+	bigquery2.Init()
 
-// VerifyConnection test connection for big query
-func (*TestHandle) VerifyConnection() error {
-	credentials, err := bigqueryCredentials()
-	if err != nil {
-		return err
-	}
+	credentials, err := testhelper.BigqueryCredentials()
+	require.NoError(t, err)
 
-	err = testhelper.WithConstantBackoff(func() (err error) {
-		handle.DB, err = bigquery2.Connect(context.TODO(), &credentials)
-		if err != nil {
-			err = fmt.Errorf("could not connect to warehouse bigquery with error: %s", err.Error())
-			return
-		}
-		return
-	})
-	if err != nil {
-		return fmt.Errorf("error while running test connection for bigquery with err: %s", err.Error())
-	}
-	return nil
-}
+	db, err := bigquery2.Connect(context.TODO(), &credentials)
+	require.NoError(t, err)
+
+	schema := testhelper.Schema(warehouseutils.BQ, testhelper.BigqueryIntegrationTestSchema)
```

Comment:
This schema is status right ? What if two tests suites run in parallel? Can we use a parallel schema every time ?

Reply:
We can't use parallel schema.

Reply:
I am having the assumption that there will only be one Integration test per destination.